### PR TITLE
newsflash: init at 1.0.1

### PR DESCRIPTION
**Pull request by kira-bruneau**
###### Motivation for this change
Adds NewsFlash. A modern feed reader designed for the GNOME desktop, and the spiritual successor to FeedReader.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/feedreaders/newsflash/default.nix [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  src = fetchFromGitLab {
+    owner = "news-flash";
+    repo = "news_flash_gtk";
+    rev = version;
+    sha256 = "1y2jj3z08m29s6ggl8q270mqnvdwibs0f2kxybqhi8mya5pyw902";
+  };
+
+  cargoPatches = [
+    ./cargo.lock.patch
+  ];
```
Comment (danieldk) [inline pkgs/applications/networking/feedreaders/newsflash/default.nix:20]: Could you open an issue upstream asking if they could add a lock file and add a comment here with a link to the issue?

**Comment by dasj19** [comment]
I tested with `nix-shell -p nixpkgs-review --run "nixpkgs-review pr 93622"` and it builds fine. However I could not find an executable to run.

**Comment by MetaDark** [comment]
@dasj19 The executable is named `com.gitlab.newsflash`

File: pkgs/applications/networking/feedreaders/newsflash/default.nix (hunk 1 above)
Comment (MetaDark) [reply pkgs/applications/networking/feedreaders/newsflash/default.nix:20]: :+1: Will do

**Comment by MetaDark** [comment]
@danieldk I've opened an issue asking upstream to add the lock file: https://gitlab.com/news-flash/news_flash_gtk/-/issues/112.

**Comment by dasj19** [comment]
> @dasj19 The executable is named `com.gitlab.newsflash`

Yes, that works fine... You can thick the 3rd checkbox in the description ("Tested compilation of all pkgs that depend on this change using...")

**Review by danieldk — APPROVED** [review]
LGTM

Result of `nixpkgs-review pr 93622` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- newsflash
</details>

I took the liberty to run `nixpkgs-fmt` on the derivation to standardize formatting of the argument attrset a bit.

**Comment by MetaDark** [comment]
@danieldk Thanks! I didn't know about `nixpkgs-fmt`, but I will use it in all my commits now.

**Pull request merged**